metadata/db.go

```diff
@@ -0,0 +1,180 @@
+package metadata
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/containerd/containerd/gc"
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+)
+
+type DB struct {
+	db *bolt.DB
+
+	rlock sync.RWMutex
```

Comment:
Let's document the locking semantics. The behavior of these is really important for implementing this safely.